hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/validate/types/HazelcastInferTypes.java

```diff
@@ -80,6 +82,10 @@
     };
 
     private HazelcastInferTypes() {
+        // No-op.
     }
 
+    public static SqlOperandTypeInference explicit(SqlTypeName typeName) {
```

Comment:
It might make sense to create a test for it in `HazelcastInferTypesTest`.

Reply:
Created a ticket for that: https://github.com/hazelcast/hazelcast/issues/17360
There is a high level of confidence that it works as expected because there are integration tests that cover this